sparse/core.py

```diff
@@ -379,6 +382,52 @@ def elemwise(self, func, *args, **kwargs):
         return COO(self.coords, func(self.data, *args, **kwargs),
                    shape=self.shape, has_duplicates=self.has_duplicates)
 
+    def elemwise_binary(self, func, other, *args, **kwargs):
+        assert isinstance(other, COO)
+        if kwargs.pop('check', True) and func(0, 0, *args, **kwargs) != 0:
+            raise ValueError("Performing this operation would produce "
+                    "a dense result: %s" % str(func))
+        if self.shape != other.shape:
+            raise ValueError("Broadcasting is not supported")
+        self.sum_duplicates()  # TODO: document side-effect or make copy
+        other.sum_duplicates()  # TODO: document side-effect or make copy
+
+        # Sort self.coords in lexographical order using record arrays
```

Comment:
Does `sum_duplicates` do this already?  If not then should it be made to?  The `sum_duplicates` method was changed a short while before you submitted this PR and now calls np.lexsort on coords directly.  It doesn't call it on a recarray though.  Honestly I didn't seriously look into how lexsort operates so am not very informed here.